### ISSUE TITLE
Change Tests to Not Store Expected Errors in the Arrange Section

### ISSUE THREAD

**Issue by InsertCreativityHere** [icerpc/slicec#205]
Lets take this test as an example:
```
        // Arrange
        let slice = "
            encoding = 1;
            module Test;
            struct A {}
        ";
        let expected_errors = [
            "struct `A` is not supported by the Slice1 encoding",
            "file encoding was set to Slice1 here:",
            "structs must be `compact` to be supported by the Slice1 encoding",
        ];

        // Act
        let error_reporter = parse_for_errors(slice);

        // Assert
        assert_errors!(error_reporter, expected_errors);
    }
```
We 'arrange' 2 things:
 1) A string that holds the slice we want to test. This is correctly placed in the arrange section.
 2) An array of errors that we expect to receive from calling the MUT. **_I think this doesn't belong in the arrange section, and should be moved to the assert section._**

 Per Microsoft: (https://docs.microsoft.com/en-us/visualstudio/test/unit-test-basics?view=vs-2022#write-your-tests)

>The Arrange section of a unit test method initializes objects and sets the value of the data that is passed to the method under test.

`slice` is being passed into the MUT, hence it belongs in the 'arrange'. `expected_errors` is not being passed into the MUT, hence it belongs in the assert.

Even putting unit testing aside, this style of test setup violates https://wiki.c2.com/?DeclareVariablesAtFirstUse.
We don't use `expected_errors` until the bottom of the function, so there's no reason to declare it at the top. It should be declared where it's used; in the assert section.

## TL;DR

Objects used exclusively for asserting should be declared in the assert section, instead of the arrange section.
Arrange should be reserved exclusively for setting up data to pass in the MUT. This keeps the focus clear.
